BinaryQueries.cpp

```diff
@@ -1,21 +1,41 @@
 #include <iostream>
 using namespace std;
+void fastscan(int &number)      //function to improve running time takes input faster
```

Comment:
I'd say its pretty dangerous to use these. scanf should be fast enough for every type of problem. Just a heads up.

Reply:
The thing is scanf() is faster than cin but getchar() is faster than scanf() and getchar_unlocked() is faster than getchar(), hence fastest of all. It's true it is not thread safe so should not be used in our systems but it is pretty good when it comes to online competitive coding as running time gets crucial there and it doesn't causes any problem whatsoever in that environment.

Reply:
CP uses single threads. So yeah, cool.